lib/editors.js

```diff
@@ -608,106 +694,39 @@ class CourseInstanceCopyEditor extends Editor {
         this.description = `Copy course instance ${this.course_instance.short_name}`;
     }
 
-    getNextNameShort(name) {
-        let found = name.match(new RegExp(`^([A-Za-z]{2})([0-9]{2})$`));
-        debug(`getNextNameShort:\n${found}`);
-        if (found) {
-            const seasons = ['Sp', 'Su', 'Fa'];
-            for (let i = 0; i < 3; i++) {
-                if (found[1] == seasons[i]) {
-                    if (i == 2) return `${seasons[0]}${(parseInt(found[2]) + 1).toString().padStart(2, '0')}`;
-                    else return `${seasons[i + 1]}${parseInt(found[2]).toString().padStart(2, '0')}`;
-                }
-            }
-        }
-        return '';
-    }
-
-    getNextNameLong(name) {
-        let found = name.match(new RegExp(`^([A-Za-z]+)([0-9]{4})$`));
-        debug(`getNextNameLong:\n${found}`);
-        if (found) {
-            const seasons = ['Spring', 'Summer', 'Fall'];
-            for (let i = 0; i < 3; i++) {
-                if (found[1] == seasons[i]) {
-                    if (i == 2) return `${seasons[0]}${(parseInt(found[2]) + 1).toString().padStart(2, '0')}`;
-                    else return `${seasons[i + 1]}${parseInt(found[2]).toString().padStart(2, '0')}`;
-                }
-            }
-        }
-        return '';
-    }
-
-    getPrefixAndNumber(name) {
-        let found = name.match(new RegExp('^(?<prefix>.*)_copy(?<number>[0-9]+)$'));
-        debug(`getPrefixAndNumber:\n${found}`);
-        if (found) {
-            return {
-                'prefix': found.groups.prefix,
-                'number': parseInt(found.groups.number),
-            };
-        } else {
-            return null;
-        }
-    }
-
     write(callback) {
         debug('CourseInstanceCopyEditor: write()');
         const courseInstancesPath = path.join(this.course.path, 'courseInstances');
         async.waterfall([
             (callback) => {
-                debug(`Generate unique short_name in ${courseInstancesPath}`);
+                debug('Get all existing long names and generate a number');
+                sqldb.query(sql.select_course_instances_with_course, {course_id: this.course.id}, (err, result) => {
+                    if (ERR(err, callback)) return;
+                    this.oldNamesLong = _.map(result.rows, 'long_name');
+                    this.order_by = (_.max(_.map(result.rows, 'order_by')) || 1) + 1;
```

Comment:
```suggestion
                    this.number = (_.max(_.map(result.rows, 'number')) || 1) + 1;
```

Reply:
good catch, my mistake (and because of the coalesce, it would never have resulted in a visible error)